lib/utils.ts

```diff
@@ -1,10 +1,19 @@
 import { type ClassValue, clsx } from "clsx"
 import { twMerge } from "tailwind-merge"
+import groupBy from 'lodash/groupBy'
  
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs))
 }
 
 export const typedObjectEntries = <TKey extends string, TValue>(input: Record<TKey, TValue>): [TKey, TValue][] => {
   return Object.entries(input) as [TKey, TValue][];
+}
+
+export const typedObjectFromEntries = <TKey extends string, TValue>(input: [TKey, TValue][]): Record<TKey, TValue> => {
+  return Object.fromEntries(input) as Record<TKey, TValue>;
+}
+
+export const typedGroupBy = <TKey extends string, TValue>(values: TValue[], groupingFunction: (value: TValue) => TKey): Record<TKey, TValue[]> => {
+  return groupBy(values, groupingFunction) as Record<TKey, TValue[]>
 }
```

Comment:
My types are more correct than the library ones in these instances even though it needs a cast. This will matter once we have a non-`any` type on `filteredData` which I'd like to do using a graphql typegen library at a later date :)